core/dbt/events/functions.py

```diff
@@ -1,125 +1,148 @@
 import betterproto
-import io
+from dbt.constants import METADATA_ENV_PREFIX
+from dbt.events.base_types import BaseEvent, Cache, EventLevel, NoFile, NoStdOut
+from dbt.events.eventmgr import EventManager, LoggerConfig, LineFormat, NoFilter
+from dbt.events.helpers import env_secrets, scrub_secrets
+from dbt.events.proto_types import EventInfo
+from dbt.events.types import EmptyLine
+import dbt.flags as flags
+from dbt.logger import GLOBAL_LOGGER, make_log_dir_if_missing
+from functools import partial
 import json
-import logging
 import os
 import sys
-import threading
+from typing import Callable, Dict, Optional, TextIO
 import uuid
-from collections import deque
-from datetime import datetime
-from io import StringIO, TextIOWrapper
-from logging import Logger
-from logging.handlers import RotatingFileHandler
-from typing import Callable, Dict, List, Optional, Union
 
-import dbt.flags as flags
-import logbook
-from colorama import Style
-from dbt.constants import METADATA_ENV_PREFIX, SECRET_ENV_PREFIX
-from dbt.events.base_types import BaseEvent, Cache, NoFile, NoStdOut
-from dbt.events.types import EmptyLine, EventBufferFull, MainReportVersion
-from dbt.logger import make_log_dir_if_missing
 
-# create the module-globals
-LOG_VERSION = 2
-EVENT_HISTORY = None
+LOG_VERSION = 3
+metadata_vars: Optional[Dict[str, str]] = None
 
-DEFAULT_FILE_LOGGER_NAME = "default_file"
-FILE_LOG = logging.getLogger(DEFAULT_FILE_LOGGER_NAME)
 
-DEFAULT_STDOUT_LOGGER_NAME = "default_std_out"
-STDOUT_LOG = logging.getLogger(DEFAULT_STDOUT_LOGGER_NAME)
+def setup_event_logger(log_path: str, log_format: str, use_colors: bool, debug: bool):
+    cleanup_event_logger()
+    make_log_dir_if_missing(log_path)
 
-invocation_id: Optional[str] = None
-metadata_vars: Optional[Dict[str, str]] = None
+    if flags.ENABLE_LEGACY_LOGGER:
+        EVENT_MANAGER.add_logger(_get_logbook_log_config(debug))
+    else:
+        EVENT_MANAGER.add_logger(_get_stdout_config(log_format, debug, use_colors))
+
+        if _CAPTURE_STREAM:
+            # Create second stdout logger to support test which want to know what's
+            # being sent to stdout.
+            capture_config = _get_stdout_config(log_format, debug, use_colors)
+            capture_config.output_stream = _CAPTURE_STREAM
+            EVENT_MANAGER.add_logger(capture_config)
+
+        # create and add the file logger to the event manager
+        EVENT_MANAGER.add_logger(
+            _get_logfile_config(os.path.join(log_path, "dbt.log"), use_colors, log_format)
+        )
 
 
-def setup_event_logger(log_path, log_format, use_colors, debug):
-    global FILE_LOG
-    global STDOUT_LOG
+def _get_stdout_config(log_format: str, debug: bool, use_colors: bool) -> LoggerConfig:
```

Comment:
@peterallenwebb  I feel like I got to a sub optimal solution for this entire file to make mypy happy. Feel free to update or point me to the right direction